From fb4f9d860bcd6bce7e0dd47bfebc4f8ec2c8b6a5 Mon Sep 17 00:00:00 2001
From: Jianqun Xu <jay.xu@rock-chips.com>
Date: Thu, 11 Oct 2018 15:24:37 +0800
Subject: [PATCH] pinctrl: rockchip: fix rk1808_mux_route_data

Fix rk1808_mux_route_data, support:
i2c2m0_sda
i2c2m1_sda
uart1_rxm0
uart1_rxm1
uart2_rxm0
uart2_rxm1
uart2_rxm2

Tested on RK1808-evb.

Change-Id: Iea221a5cab8a63e336e183d30f3faaffb3a14873
Signed-off-by: Jianqun Xu <jay.xu@rock-chips.com>
---
 drivers/pinctrl/pinctrl-rockchip.c | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/drivers/pinctrl/pinctrl-rockchip.c b/drivers/pinctrl/pinctrl-rockchip.c
index 429fad356119..a9a7e9b4289f 100644
--- a/drivers/pinctrl/pinctrl-rockchip.c
+++ b/drivers/pinctrl/pinctrl-rockchip.c
@@ -935,6 +935,20 @@ static struct rockchip_mux_route_data rk1808_mux_route_data[] = {
 		.func = 2,
 		.route_offset = 0x190,
 		.route_val = BIT(16 + 3) | BIT(3),
+	}, {
+		/* uart1_rxm0 */
+		.bank_num = 4,
+		.pin = 8,
+		.func = 2,
+		.route_offset = 0x190,
+		.route_val = BIT(16 + 13),
+	}, {
+		/* uart1_rxm1 */
+		.bank_num = 1,
+		.pin = 12,
+		.func = 3,
+		.route_offset = 0x190,
+		.route_val = BIT(16 + 13) | BIT(13),
 	}, {
 		/* uart2_rxm0 */
 		.bank_num = 4,
@@ -948,7 +962,7 @@ static struct rockchip_mux_route_data rk1808_mux_route_data[] = {
 		.pin = 25,
 		.func = 2,
 		.route_offset = 0x190,
-		.route_val = BIT(16 + 14) | BIT(14) | BIT(16 + 15),
+		.route_val = BIT(16 + 14) | BIT(16 + 15) | BIT(14),
 	}, {
 		/* uart2_rxm2 */
 		.bank_num = 3,
-- 
2.35.3

